### PR TITLE
Make nice-to-have improvements for the Facilitator Application, and

### PR DESCRIPTION
**Pull request by aoby**
add UsPhoneNumberInput control with storybook and unit tests.

The `UsPhoneNumberInput` control looks like this:
image

It only accepts input in phone number format. It's relatively friendly though and tries to coerce what is actually entered into the right format.
image

Format validation failures look like this:
image
and
image

### CODE REVIEW

File: apps/src/code-studio/pd/form_components/FormComponent.jsx [hunk 1]
```diff
@@ -231,11 +258,21 @@ export default class FormComponent extends React.Component {
...
   }
+
+  /**
+   * Override in derived classes
+   * @param {Object} data - form data for this page
+   * @returns {Object} - custom error messages per field
+   */
+  static getErrorMessages(data) {
+    return {};
+  }
```
Comment (aoby) [inline apps/src/code-studio/pd/form_components/FormComponent.jsx:269]: CC @Hamms I added the ability to supply custom error messages and fail validation (i.e. for bad format). See screenshots and example usage above for phone number and zip.

**Comment by aoby** [comment]
[Facilitator App: post bug bash clean up](https://trello.com/c/nLo56gY0/178-facilitator-app-post-bug-bash-clean-up)

File: apps/src/code-studio/pd/application/facilitator1819/Facilitator1819Labels.jsx [hunk 2]
```diff
@@ -19,7 +19,8 @@ const pageLabels = {
     institutionType: "What type of institution do you work for?",
     currentEmployer: "Current employer",
     jobTitle: "What is your job title?",
-    resumeLink: "Please provide a link to your resume, LinkedIn profile, website, or summarize your relevant past experience.",
+    resumeLink:
```
Comment (mehalshah) [inline apps/src/code-studio/pd/application/facilitator1819/Facilitator1819Labels.jsx:22]: Not now - but we need to fix this later. We should ask for resume uploads I think

File: apps/src/code-studio/pd/form_components/UsPhoneNumberInput.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    this.state = {
+      value: this.props.value
+    };
+  }
+
+  static isValid(value) {
+    return /^\d{10}$/.test(value);
+  }
+
```
Comment (mehalshah) [inline apps/src/code-studio/pd/form_components/UsPhoneNumberInput.jsx]: comment on what this is doing

File: apps/src/code-studio/pd/application/facilitator1819/Section1QuestionsAboutYou.jsx [hunk 4]
```diff
@@ -118,5 +122,22 @@ export default class Section1QuestionsAboutYou extends Facilitator1819FormCompon
...
+   */
+  static getErrorMessages(data) {
+    const formatErrors = {};
+
+    if (data.zipCode && !ZIP_CODE_REGEX.exec(data.zipCode)) {
+      formatErrors.zipCode = "Must be a valid zip code";
+    }
+
+    if (!UsPhoneNumberInput.isValid(data.phone)) {
+      formatErrors.phone = "Must be a valid US phone number";
```
Comment (mehalshah) [inline apps/src/code-studio/pd/application/facilitator1819/Section1QuestionsAboutYou.jsx]: Suggest making this "Must be a valid area code and phone number"

Also we should figure out if we are going to allow extensions

Comment (aoby) [reply apps/src/code-studio/pd/application/facilitator1819/Section1QuestionsAboutYou.jsx]: "area code and phone number" or "10 digit phone number" ?

We may add extensions later, but it's lower priority.

**Pull request merged**